### ISSUE TITLE
WP Admin dashboard administration

### ISSUE THREAD

**Issue by polevaultweb** [GlotPress/GlotPress#14]
Add GlotPress menu. Ability to administer projects, translation sets etc.

**Comment by ocean90** [comment]
Maybe we should dicuss first which options would be needed and if it should happen in admin or on front end.

**Comment by polevaultweb** [comment]
Personally I think the majority of the admin related stuff makes more sense in the wp admin. The wip PR will replicate admin, without removing from the frontend. Currently looking at projects admin.

**Comment by bradt** [comment]
This is waaay too lofty of an undertaking for this milestone.

**Comment by polevaultweb** [comment]
Ha!

**Comment by bradt** [comment]
Also, what did you have in mind exactly for "Ability to administer projects, translation sets etc." ? Isn't all that already available from the GlotPress UI? Wouldn't it just be duplication?

**Comment by polevaultweb** [comment]
True, but if you were building the plugin from scratch as a WP plugin, wouldn't you utilise the WP admin, tables, UI and functionality to manage adding, editing projects etc?

**Comment by polevaultweb** [comment]
It's only available in the frontend because GP has no backend

**Comment by ocean90** [comment]
Projects have only 6 settings. Once they're set you'll probably never change them again.

Even as a plugin, we should try to use WordPress just as a library. I think it should be possible that you can have people who can access GlotPress without knowing that WordPress is actually running behind it.

>  utilise the WP admin, tables, UI and functionality to manage adding, editing projects etc

If you can make it work for ~40k projects which can be three levels deep, sure. But I bet that's not that easy. :)

**Comment by polevaultweb** [comment]
Good points. Will park it for now

**Comment by toolstack** [comment]
The one setting we do need to add is the ability to change the slug used for GlotPress's URL.  At the moment it's set to "/glotpress", but a site may want something different.

**Comment by ocean90** [comment]
@toolstack Such a setting can be added to the existing Permalinks screen. Or make it filterable. :)

**Comment by toolstack** [comment]
@ocean90 could be, but that hiding it a bit, the settings menu doesn't have to a top level menu, it could be in the settings menu.  I would think many installs will want to change the link, before it was easy as you just changed the directory you installed in to, but as a plugin that's more difficult.

**Comment by bradt** [comment]
@toolstack It is already configurable via the `GP_URL_BASE` constant.

**Comment by toolstack** [comment]
There's configurable and then there's easily configurable :)

The plugin gives us an easy interface to let the admin configure this kind of setting without messing with the config file.

**Comment by A5hleyRich** [comment]
It seems overkill adding a settings screen just to define the base URL. Often it's a one time change so the menu item would become redundant.
